hierarchicalforecast/methods.py

```diff
@@ -508,6 +541,7 @@ def reconcile(self,
         `y_hat`: Forecast values of size (`base`, `horizon`).<br>
         `y_insample`: Insample values of size (`base`, `insample_size`). Only used by `wls_var`, `mint_cov`, `mint_shrink`<br>
         `y_hat_insample`: Insample fitted values of size (`base`, `insample_size`). Only used by `wls_var`, `mint_cov`, `mint_shrink`<br>
+        `idx_bottom`: Indices corresponding to the bottom level of `S`, size (`bottom`).<br>
```

Comment:
Would it be possible to infer the idx_bottom indexes from the S_[a+b, b] matrix?
something like  `idx_bottom = n.arange(a+b)[a:]`

If we commit to ordered numpy arrays, things might simplify.

Additionally the PERMBU method is operating under the assumption of hierarchically ordered numpy arrays.

Reply:
It's possible, but it's just safer not to infer them. Also, it's not obvious how to test the assumption of hierarchically ordered numpy arrays. The complexity inside the methods simplifies the user experience in the `HierarchicalReconciliation` class.